localstack/config.py

```diff
@@ -32,6 +32,115 @@
 load_start_time = time.time()
 
 
+class Directories:
+    """
+    Holds the different directories available to localstack. Some directories are shared between the host and the
+    localstack container, some live only on the host and some only in the container.
+
+    Attributes:
+        infra:      container only; static infrastructure: binaries and libraries packaged with the image
+        var:        shared; variable data computed at runtime: lazy-loaded binaries, ssl cert, ...
+        cache:      shared; ephemeral data that has to persist across localstack runs and reboots
+        tmp:        individual; ephemeral data that does not have to persist across localstack runs but not reboots
+        shared_tmp: shared; ephemeral data used to communicate between container<->lambdas
+        data:       shared; holds localstack state, pods, ...
+        config:     host only; pre-defined configuration values, cached credentials, machine id, ...
+        init:       shared; user-defined provisioning scripts executed when in the container when it starts
+        logs:       shared; log files produced by localstack
+    """
+
+    infra: str
+    var: str
+    cache: str
+    tmp: str
+    shared_tmp: str
+    data: str
+    config: str
+    init: str
+    logs: str
+
+    def __init__(
+        self,
+        infra: str = None,
+        var: str = None,
+        cache: str = None,
+        tmp: str = None,
+        shared_tmp: str = None,
+        data: str = None,
+        config: str = None,
+        init: str = None,
+        logs: str = None,
+    ) -> None:
+        super().__init__()
+        self.infra = infra
+        self.var = var
+        self.cache = cache
+        self.tmp = tmp
+        self.shared_tmp = shared_tmp
+        self.data = data
+        self.config = config
+        self.init = init
+        self.logs = logs
+
+    @staticmethod
+    def from_config():
+        """Returns Localstack directory paths from the config/environment variables defined by the config."""
+        return Directories(
+            infra=INSTALL_DIR_INFRA,
+            var=TMP_FOLDER,  # TODO: add variable
+            cache=TMP_FOLDER,  # TODO: add variable
+            tmp=TMP_FOLDER,  # TODO: move default value to /tmp/localstack/host
+            shared_tmp=HOST_TMP_FOLDER,  # TODO: move default value to /tmp/localstack/shared
```

Comment:
Not sure about this one - `HOST_TMP_FOLDER` is currently only the mapped folder from the host (required for mounting code into spawned Lambda containers). There should not be a need for it to be accessed or created directly inside the container (specifically, should not be created in `mkdirs()` in line 120 below). It is currently only used for setting the volume mount flag (`-v ...:...`) when spawning Lambda containers, with `LAMBDA_REMOTE_DOCKER=false` enabled.

Maybe this is just a matter of nomenclature - the `shared_` prefix already indicates that this is one end of a shared folder. Maybe we just need to make it a bit more clear that this is the external (on the host) part of the shared tmp (e.g., `shared_tmp_host`), whereas the current `tmp` folder is the corresponding internal part of the shared folder.

Reply:
how about we make this explicitly a `lambda` dir, that's shared only between the host and lambdas? (see my comment below)